e2e/js_image/runfiles.bzl

```diff
@@ -0,0 +1,129 @@
+"contains container helper functions for js_binary"
+
+load("@rules_pkg//:providers.bzl", "PackageFilegroupInfo", "PackageFilesInfo", "PackageSymlinkInfo")
+load("@rules_pkg//:pkg.bzl", "pkg_tar")
+
+def _runfile_path(ctx, file, runfiles_dir):
+    path = file.short_path
+    if path.startswith(".."):
+        return path.replace("..", runfiles_dir)
+    if not file.owner.workspace_name:
+        return "/".join([runfiles_dir, ctx.workspace_name, path])
+    return path
+
+_LAUNCHER_TMPL = """
```

Comment:
nit: I think you could toss this whole launcher if you just put `env = {"BAZEL_BINDIR": "."}` on the binary. We did that for linters in RH.

Reply:
that was the initial solution but Greg suggested that it would be a leaky implementation detail. You'd need to what this is and put this into env.  Also, `args` and `env` is not propagated through DefaultInfo so i am not sure how this would work with binary.